### PR TITLE
[TTAHUB-2493] Disable objective title if linked to an approved report

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

Don't allow editing of objective title if its linked to a approved report.

## How to test

Edit a draft report that is using a goal/objective that is also used by an approved report. The objective title should not be editable.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2493

### Every PR

- [x] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/ActivityReport/Pages/components/Objective.js [hunk 1]
```diff
@@ -225,15 +224,9 @@ export default function Objective({
     rules: { required: true },
     defaultValue: objective.closeSuspendContext || '',
   });
+  const isOnApprovedReport = objective.onApprovedAR;
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/ActivityReport/Pages/components/Objective.js:227]: Not sure if we need a useEffect on theses.

Comment (thewatermethod) [reply frontend/src/pages/ActivityReport/Pages/components/Objective.js:227]: you do not, no

**Comment by thewatermethod** [comment]
What you have is good, but this isn't enough to make it work - the backend needs to be adjusted as well, specifically you need to add 
```js
    'onApprovedAR',
     'onAR',
 ```
 to lines src/goalServices/goals.js:845
 
 Then, the TTA provided should always be editable on an AR. You can change
 frontend/src/pages/ActivityReport/Pages/components/Objective.js:406 to

```js
isOnApprovedReport={false}
```

**Comment by thewatermethod** [comment]
I am unable to change topics on an existing objective

**Comment by AdamAdHocTeam** [comment]
> I am unable to change topics on an existing objective

This has been fixed also the merging of existing used topics with new topics.

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**